### ISSUE TITLE
Regexp request header doesn't seem to match

### ISSUE THREAD

**Issue by georgepapas** [DiUS/pact-jvm#213]
See this repo for tests that fail on the regexp header matcher...

https://github.com/georgepapas/pact-issue-212

Am I missing something here?

**Comment by georgepapas** [comment]
Looks like there is some difference between 3.1.3 and 3.2.3

in 3.1.3 this form of regex works

`headers: ['Accept': regexp(~'.*application/json.*', 'application/json')]`

but in 3.2.3 it doesnt

Seems like this doesn't work in 3.2.3 or 3.1.3

`headers: ['Accept': regexp(~/.*application/json.*/, 'application/json')]`

**Comment by uglyog** [comment]
It is failing for me due to the charset on the Content-Type header:

    HeaderMismatch - Expected header 'Content-Type' to have value 'application/json' but was 'application/json;charset=UTF-8'

Setting the correct mimeType fixes the tests:

```groovy
      withBody(mimeType: 'application/json;charset=UTF-8') {
        priceInCents 'x0x'
      }
```

This is something I want to fix so that a missing charset does not cause a failure

**Comment by georgepapas** [comment]
Thanks

**Comment by uglyog** [comment]
Are you getting a different failure?

**Comment by georgepapas** [comment]
Mime type got me passed the issue i was experiencing.  I am seeing though multiple tests to the same expected endpoint/method dont behave correctly.  Looks like state is being retained by the mock server between tests?

**Comment by uglyog** [comment]
this has been released with v 3.2.4
